frame/contracts/src/storage/meter.rs

```diff
@@ -406,6 +406,16 @@ where
 		};
 	}
 
+	/// Adds a deposit charge.
+	///
+	/// Use this method instead of [`Self::charge`] when the charge is not the result of a storage
+	/// change. This is the case when a `delegate_dependency` is added or removed, or when the
+	/// `code_hash` is updated.
+	pub fn charge_deposit(&mut self, deposit_account: DepositAccount<T>, amount: DepositOf<T>) {
```

Comment:
isn't that what I am doing here 
```rust
Use this method instead of [`Self::charge`] when the charge is not the result of a storage
	/// change
```

Reply:
But why can't `charge` be used in this case?